mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp

```diff
@@ -125,12 +125,53 @@ struct MIIdRewritePattern : public OpRewritePattern<Tmi> {
 struct MIMFMARewritePattern : public OpRewritePattern<miopen::MFMAV2Op> {
   using OpRewritePattern<miopen::MFMAV2Op>::OpRewritePattern;
 
+  // The below implements the concatenation of 4 i8 to an i32:
+  // a[0] | (a[1] << 8) | (a[2] << 16) | (a[3] << 24)
+  void vector4i8Toi32(const Value &vec4i8, Value &i32vals,
+                      PatternRewriter &b) const {
+    auto loc = vec4i8.getLoc();
+    for (int i = 0; i < 4; ++i) {
+      Value iterOp = b.create<arith::ConstantIndexOp>(loc, i);
+      Value extracted = b.create<vector::ExtractElementOp>(
+          loc, b.getIntegerType(8), vec4i8, iterOp);
+      Value i32A =
+          b.create<arith::ExtUIOp>(loc, extracted, b.getIntegerType(32));
+      Value shiftWidth =
+          b.create<arith::ConstantIntOp>(loc, i * 8, b.getIntegerType(32));
+      Value i32AShifted = b.create<arith::ShLIOp>(loc, i32A, shiftWidth);
+      i32vals = b.create<arith::OrIOp>(loc, i32vals, i32AShifted);
+    }
+  }
+
   LogicalResult matchAndRewrite(miopen::MFMAV2Op op,
                                 PatternRewriter &b) const override {
-    auto gpuMfmaOp = b.create<gpu::MFMAOp>(
-        op.getLoc(), op.getType(), op.sourceA(), op.sourceB(), op.destC());
+    auto loc = op.getLoc();
+    bool isVeci8 = false;
+    if (op.sourceA().getType().isa<VectorType>()) {
+      auto vectorType = op.sourceA().getType().cast<VectorType>();
+      isVeci8 = vectorType.getElementType() == b.getIntegerType(8);
+    }
+
+    gpu::MFMAOp gpuMfmaOp;
+    // Note: For i8 type, gpu.mfma op requires both arguments to be i32 instead
+    // of vector<4xi8>, thus the explicit conversion below
+    if (isVeci8) {
+      Value sourceA =
+          b.create<arith::ConstantIntOp>(loc, 0, b.getIntegerType(32));
+      Value sourceB =
+          b.create<arith::ConstantIntOp>(loc, 0, b.getIntegerType(32));
+      vector4i8Toi32(op.sourceA(), sourceA, b);
+      vector4i8Toi32(op.sourceB(), sourceB, b);
+      gpuMfmaOp = b.create<gpu::MFMAOp>(op.getLoc(), op.getType(), sourceA,
+                                        sourceB, op.destC());
+    } else {
+      gpuMfmaOp = b.create<gpu::MFMAOp>(op.getLoc(), op.getType(), op.sourceA(),
```

Comment:
It might be better to always have `Value sourceA = op.sourceA();` then to have the if statement to update `sourceA` and `sourceB` for `i8` vectors - that wey we don't repeat the creation of the MFMA Op

Reply:
Sounds reasonable, will do

Reply:
Addressed in d1a2fab